### DISCUSSION TITLE
Prisma db pull with Supabase (Postgresql)

### DISCUSSION THREAD

**Discussion by idiglove** [prisma/prisma#11190] [Q&A] [answered]
Hi, I'm having a problem where after a successful db pull and modifying columns in supabase directly, another db pull results to this error:

```
Introspecting based on datasource defined in prisma/schema.prisma …

Error: Error in connector: Error querying the database: Error querying the database: Error querying the database: db error: ERROR: prepared statement "s0" already exists
```

I've been looking around the internet for an answer, and I'm just blocked. Would be glad to get an answer or any help for this,
cheers!

**Comment by janpio** [comment]
Which of the two connection strings Supabase offers are you using? For Introspection or Migration, you should use the non connection pooled one.

**Reply by idiglove** [reply]
Introspection. It works now after I removed `?schema=public` and added `pgbouncer=true` 
Can you help explain what happened there?

**Reply by janpio** [reply, marked as answer]
It seems you are using the pooled connection string then which runs behind PgBouncer. That unfortunately requires the manual additional for `pgbouncer=true` in any case. But be aware Introspection and Migration might still fail with that connection string - you should use the other, non pooled one for these use cases.

**Reply by idiglove** [reply]
Ok got it, thank you!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
